generators/app/templates/sources/main/helpers/analytics/analytics.service.ts

```diff
@@ -0,0 +1,76 @@
+import app from 'main.module';
+import {ILogger, LoggerService} from 'helpers/logger/logger';
+import {IApplicationConfig} from 'main.constants';
+
+interface IWindowWithAnalytics extends ng.IWindowService {
+  googleAnalytics: any;
+}
+
+/**
+ * Analytics service: insert Google Analytics library in the page.
+ */
+export class AnalyticsService {
+
+  private logger: ILogger;
+  private analyticsAreActive = false;
+
+  constructor(private $window: IWindowWithAnalytics,
+              private config: IApplicationConfig,
+              logger: LoggerService) {
+
+    this.logger = logger.getLogger('analyticsService');
+
+    this.init();
```

Comment:
I believe the responsibility of customizing the creation of the GA object lies within the analytics service. All the Cordova specific code can be added in analytics-service instead of main-run.

Regarding disabling of analytics in debug, there are some cases where having them on is useful (typically when one wants to test out a newly-added event tracker). Of course you would want a different analytics account for debugging and for actual production, but that is just a matter of setting up the gulp-replace task around IApplicationConfig.googleAnalyticsId. If you want to disable them entirely in debug mode, it can also be done the same way, by setting the value to null in debug mode.